# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/filedialog/PWinFileDialog.ipdl
# Commit: d56484304ec3
# Full Hash: d56484304ec3abe3891ede9e8d141c95543d3e37
# Author: Ray Kraesig <rkraesig@mozilla.com>
# Date: 2023-09-16 09:14:45
# Regressor Bug: 1837079
# File Overlap Count: 2
# Description:
#   Bug 1837079 - [5/10] Create WinFileDialog IPDL protocol and implementation  r=gstoll,handyman,ipc-reviewers,nika
#   
#   Create and implement a new top-level IPC protocol, `PWinFileDialog`,
#   using the primitives from bug 1833450 and the new sandboxing type from
#   the previous commit.
# ==============================================================================

diff -r 36d3afd881b4 -r d56484304ec3 widget/windows/filedialog/PWinFileDialog.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/widget/windows/filedialog/PWinFileDialog.ipdl	Fri Sep 15 18:10:01 2023 +0000
@@ -0,0 +1,31 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set sw=2 ts=8 et ft=ipdl : */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include WinFileDialogCommandsDefn;
+using mozilla::WindowsHandle from "mozilla/ipc/IPCTypes.h";
+using mozilla::widget::filedialog::FileDialogType from "mozilla/widget/filedialog/WinFileDialogCommands.h";
+
+namespace mozilla {
+namespace widget {
+namespace filedialog {
+
+protocol PWinFileDialog {
+
+child:
+  // Exactly one Show function should be called per instance. Further calls will
+  // result in IPC failure.
+  //
+  // Each will return `Nothing` iff the operation was canceled by the user.
+
+  async ShowFileDialog(WindowsHandle parentHwnd, FileDialogType type, Command[] commands)
+    returns (Results? results);
+  async ShowFolderDialog(WindowsHandle parentHwnd, Command[] commands)
+    returns (nsString? path);
+};
+
+}  // namespace filedialog
+}  // namespace widget
+}  // namespace mozilla